api/envoy/config/filter/network/redis_proxy/v2/redis_proxy.proto

```diff
@@ -35,4 +36,37 @@ message RedisProxy {
 
   // Network settings for the connection pool to the upstream cluster.
   ConnPoolSettings settings = 3 [(validate.rules).message.required = true];
+
+  message RouteMatch {
+    // If specified, the route is a prefix rule that must match the beginning of the keys. The proxy
+    // will always favor the longest matches.
+    string prefix = 1 [(validate.rules).string.min_bytes = 1];
+
+    // Indicates that prefix/path matching should be case insensitive. The default is true.
+    google.protobuf.BoolValue case_sensitive = 2;
+  }
+
+  message RouteAction {
+    string cluster = 1 [(validate.rules).string.min_bytes = 1];
+  }
+
+  message Route {
+    // Name of the route that will be used for instrumentation purposes.
+    string name = 1 [(validate.rules).string.min_bytes = 1];
+
+    // Route matching parameters.
+    RouteMatch match = 2 [(validate.rules).message.required = true, (gogoproto.nullable) = false];
+
+    // Route request to some upstream cluster.
+    RouteAction route = 3 [(validate.rules).message.required = true, (gogoproto.nullable) = false];
+
+    // Network settings for the connection pool to the upream cluster.
+    ConnPoolSettings settings = 4 [(validate.rules).message.required = true];
```

Comment:
i wonder if instead we should try to make this more closely match the API of the HTTP router, which would be to include a timeout field directly and instead of using the settings object.

regardless, think it would be good to allow use of a global default from the parent settings field so every route doesn't require its own settings field.

Reply:
Yes, that's why I opted of using a `RouteConfiguration` message to be able to set shared sane defaults for all routes. It seemed pretty similar to how the HTTP router does it.

Would you rather use the default redis_proxy.settings for all routes, or have a redis_proxy.route_config.settings be the default? The reason I'm asking this is because I've currently used the `redis_proxy.cluster` and `redis_proxy.settings` as the default route used to forward every commands that does not match any prefix.